### PR TITLE
docs(yaml): describe sequences

### PR DESCRIPTION
**Pull request by eeberhard**
## Description

- #129 

This PR solves the issue by describing the sequencing syntax introduced in application schema 1-2-0. I also re-structured the YAML syntax page to present a bit more of a logical order.

This is a draft for now until the base image release 3.1.0 is out.

## Review guidelines

Estimated Time of Review: x minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

File: docs/docs/reference/02-yaml-syntax.md [hunk 1]
```diff
@@ -23,59 +23,402 @@ hardware:
...
+
+```yaml
+transition: <component_name>
+```
+
+#### Trigger a lifecycle transition
+
+```yaml
+lifecycle: "configure"
+```
```
Comment (domire8) [inline docs/docs/reference/02-yaml-syntax.md:224]: In my humble opinion, lines 222 to 224 should be removed, its more confusing than anything else no?

File: docs/docs/reference/02-yaml-syntax.md [hunk 2]
```diff
@@ -23,59 +23,402 @@ hardware:
...
-### Position
+### Special event predicates
 
-The `position` field is used to define the desired location of the button in the application graph. It has two subfields
-defining the X and Y location, respectively.
+#### on_load
+
+The `on_load` predicate is provided by the state engine and set to true after the component
+has been loaded. Any events associated with the `on_load` predicate are handled once
+on instantiation of the node.
```
Comment (bpapaspyros) [inline docs/docs/reference/02-yaml-syntax.md:403]: nitpicking (or misreading -- excuse my English if so --). The second part reads a bit odd to me, I think a comma after `once` would fix the ambiguity in my head :P (or rephrasing).

File: docs/docs/reference/02-yaml-syntax.md [hunk 3]
```diff
@@ -264,471 +605,299 @@ for available event syntax.
...
+A sequence is a list of steps that are handled sequentially in order. Sequence steps are either
+standard [state events](#events) or conditional blocks; the conditional steps are used either to wait for a condition,
+predicate or fixed time interval, or to assert the current value of a condition or predicate.
 
-#### Set a parameter
+Similar to [conditions](#conditions), sequences are listed under a top-level field called `sequences`. Sequence names
+must be unique, and should generally follow the `lower_camel_case` naming convention.
 
-Set a parameter on the component that is triggering the event.
+After sequences are defined in the yaml, they can be managed using [sequence state events](#manage-sequences); component
```
Comment (bpapaspyros) [inline docs/docs/reference/02-yaml-syntax.md:719]: Here, should the semicolon be a comma after `sequence state events`?

Comment (bpapaspyros) [reply docs/docs/reference/02-yaml-syntax.md:719]: ah, probably not.

File: docs/docs/reference/02-yaml-syntax.md (hunk 1 above)
Comment (eeberhard) [reply docs/docs/reference/02-yaml-syntax.md:224]: Yeah true, it's a bit weird that it's up here before the text, probably a copy mistake

File: docs/docs/reference/02-yaml-syntax.md [hunk 4]
```diff
@@ -23,59 +23,402 @@ hardware:
...
 
 ```yaml
-buttons:
-  my_button:
-    position: ...
-    on_click:
-      ...
+sequence:
+  - start: "sequence_a"
+  - start: "sequence_b"
```
Comment (eeberhard) [inline docs/docs/reference/02-yaml-syntax.md:394]: single-world yaml strings are not quoted in most of the document

**Review by eeberhard — COMMENTED** [review]
Thanks for the comments! Restructuring the page a bit added a lot of extra diff but it's a great chance to read the old lines again and spot small issues. I also found a few things.

File: docs/docs/reference/02-yaml-syntax.md (hunk 2 above)
Comment (bpapaspyros) [reply docs/docs/reference/02-yaml-syntax.md:403]: The suggested version sounds clearer to me, but I agree that even the first one works.

File: docs/docs/reference/02-yaml-syntax.md (hunk 3 above)
Comment (bpapaspyros) [reply docs/docs/reference/02-yaml-syntax.md:719]: Ok, by reading again it indeed makes more sense. Then, perhaps I would personally prefer a full stop, but ok this is neither incomprehensible nor wrong, and I'm at my limit with grammar :D. Either is fine with me.

Thanks for putting all this together, it's a nice update (and sorry for the nitpicks).

**Review by domire8 — APPROVED** [review]
Thanks

**Pull request merged**